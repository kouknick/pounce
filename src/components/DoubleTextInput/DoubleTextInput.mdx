`DoubleTextInput` is used as a helper component, it groups two input components under the same authority.
Triggered events are propagated across the children components.

```typescript jsx
const Example = () => {
  const [valueFrom, setValueFrom] = React.useState('Joe');
  const [valueTo, setValueTo] = React.useState('Doe');

  return (
    <DoubleTextInput
      truncated
      labelFrom="Name"
      labelTo="Surname"
      name="details"
      from={valueFrom}
      to={valueTo}
      onChangeFrom={e => setValueFrom(e.target.value)}
      onChangeTo={e => setValueTo(e.target.value)}
    />
  );
};

<Example />;
```

This component inherits it's functionality from the `TextInput` component, thus it can be invalid or disabled.

```typescript jsx
const Example = () => {
  const [valueFrom, setValueFrom] = React.useState('Joe');
  const [valueTo, setValueTo] = React.useState('Doe');

  return (
    <DoubleTextInput
      truncated
      invalid
      labelFrom="Name"
      labelTo="Surname"
      name="details"
      from={valueFrom}
      to={valueTo}
      onChangeFrom={e => setValueFrom(e.target.value)}
      onChangeTo={e => setValueTo(e.target.value)}
    />
  );
};

<Example />;
```
