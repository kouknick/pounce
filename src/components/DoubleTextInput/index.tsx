export { default } from './DoubleTextInput';
export type { DoubleTextInputProps } from './DoubleTextInput';
