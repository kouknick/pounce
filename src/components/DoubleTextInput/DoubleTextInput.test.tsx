import React from 'react';
import { renderWithTheme } from 'test-utils';
import DoubleTextInput from './index';

const props = {
  labelFrom: 'Name',
  labelTo: 'Surname',
};

it('renders', async () => {
  const { container } = await renderWithTheme(<DoubleTextInput {...props} />);
  expect(container).toMatchSnapshot();
});

it('renders with disabled option', async () => {
  const { container } = await renderWithTheme(<DoubleTextInput {...props} disabled />);
  expect(container).toMatchSnapshot();
});

it('renders with invalid option', async () => {
  const { container } = await renderWithTheme(<DoubleTextInput {...props} invalid />);
  expect(container).toMatchSnapshot();
});

it('renders with required option', async () => {
  const { container } = await renderWithTheme(<DoubleTextInput {...props} required />);
  expect(container).toMatchSnapshot();
});

it('renders with icons', async () => {
  const { container } = await renderWithTheme(
    <>
      <DoubleTextInput {...props} icon="wrench" />
      <DoubleTextInput {...props} icon="calendar" />
      <DoubleTextInput {...props} icon="caret-down" />
    </>
  );
  expect(container).toMatchSnapshot();
});
