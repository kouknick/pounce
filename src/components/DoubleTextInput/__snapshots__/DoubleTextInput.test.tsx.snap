// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.pounce-4 {
  position: relative;
}

.pounce-16 {
  background-color: transparent;
  min-height: 47px;
  border: 1px solid;
  border-radius: 4px;
  border-color: #3e516b;
  position: relative;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-16:hover {
  border-color: #0081c5;
}

.pounce-16:disabled,
.pounce-16:disabled:focus,
.pounce-16:disabled:hover,
.pounce-16[aria-disabled=true],
.pounce-16[aria-disabled=true]:focus,
.pounce-16[aria-disabled=true]:hover {
  opacity: 0.3;
  pointer-events: none;
  cursor: default;
}

.pounce-16:focus-within {
  border-color: #0081c5;
}

.pounce-16:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.pounce-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.pounce-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  color: #f6f6f6;
  background-color: transparent;
  vertical-align: top;
  width: 100%;
  height: 100%;
  border: 0;
  position: relative;
  font-size: 0.875rem;
  font-weight: 500;
}

.pounce-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.pounce-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::placeholder {
  opacity: 0.4;
}

.pounce-0:disabled,
.pounce-0:disabled:focus,
.pounce-0:disabled:hover,
.pounce-0[aria-disabled=true],
.pounce-0[aria-disabled=true]:focus,
.pounce-0[aria-disabled=true]:hover {
  opacity: 1;
  pointer-events: none;
  cursor: default;
}

.pounce-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.pounce-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-2 {
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  left: 0;
  position: absolute;
  font-size: 0.875rem;
  font-weight: 400;
  pointer-events: none;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-16 pounce-1"
    >
      <div
        class="pounce-14 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with disabled option 1`] = `
.pounce-4 {
  position: relative;
}

.pounce-16 {
  background-color: transparent;
  min-height: 47px;
  border: 1px solid;
  border-radius: 4px;
  border-color: #3e516b;
  position: relative;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-16:hover {
  border-color: #0081c5;
}

.pounce-16:disabled,
.pounce-16:disabled:focus,
.pounce-16:disabled:hover,
.pounce-16[aria-disabled=true],
.pounce-16[aria-disabled=true]:focus,
.pounce-16[aria-disabled=true]:hover {
  opacity: 0.3;
  pointer-events: none;
  cursor: default;
}

.pounce-16:focus-within {
  border-color: #0081c5;
}

.pounce-16:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.pounce-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.pounce-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  color: #f6f6f6;
  background-color: transparent;
  vertical-align: top;
  width: 100%;
  height: 100%;
  border: 0;
  position: relative;
  font-size: 0.875rem;
  font-weight: 500;
}

.pounce-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.pounce-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::placeholder {
  opacity: 0.4;
}

.pounce-0:disabled,
.pounce-0:disabled:focus,
.pounce-0:disabled:hover,
.pounce-0[aria-disabled=true],
.pounce-0[aria-disabled=true]:focus,
.pounce-0[aria-disabled=true]:hover {
  opacity: 1;
  pointer-events: none;
  cursor: default;
}

.pounce-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.pounce-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-2 {
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  left: 0;
  position: absolute;
  font-size: 0.875rem;
  font-weight: 400;
  pointer-events: none;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      aria-disabled="true"
      class="pounce-16 pounce-1"
    >
      <div
        class="pounce-14 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-disabled="true"
            aria-invalid="false"
            class="pounce-0 pounce-1"
            disabled=""
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-disabled="true"
            aria-invalid="false"
            class="pounce-0 pounce-1"
            disabled=""
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with icons 1`] = `
.pounce-4 {
  position: relative;
}

.pounce-18 {
  background-color: transparent;
  min-height: 47px;
  border: 1px solid;
  border-radius: 4px;
  border-color: #3e516b;
  position: relative;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-18:hover {
  border-color: #0081c5;
}

.pounce-18:disabled,
.pounce-18:disabled:focus,
.pounce-18:disabled:hover,
.pounce-18[aria-disabled=true],
.pounce-18[aria-disabled=true]:focus,
.pounce-18[aria-disabled=true]:hover {
  opacity: 0.3;
  pointer-events: none;
  cursor: default;
}

.pounce-18:focus-within {
  border-color: #0081c5;
}

.pounce-18:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.pounce-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.pounce-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  color: #f6f6f6;
  background-color: transparent;
  vertical-align: top;
  width: 100%;
  height: 100%;
  border: 0;
  position: relative;
  font-size: 0.875rem;
  font-weight: 500;
}

.pounce-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.pounce-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::placeholder {
  opacity: 0.4;
}

.pounce-0:disabled,
.pounce-0:disabled:focus,
.pounce-0:disabled:hover,
.pounce-0[aria-disabled=true],
.pounce-0[aria-disabled=true]:focus,
.pounce-0[aria-disabled=true]:hover {
  opacity: 1;
  pointer-events: none;
  cursor: default;
}

.pounce-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.pounce-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-2 {
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  left: 0;
  position: absolute;
  font-size: 0.875rem;
  font-weight: 400;
  pointer-events: none;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-14 {
  margin-left: 16px;
  margin-right: 16px;
  color: currentColor;
  display: inline-block;
  vertical-align: sub;
  width: 20px;
  height: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-18 pounce-1"
    >
      <div
        class="pounce-16 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
        <svg
          class="pounce-14 pounce-1"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.09 2.90995C10.08 0.899954 7.07 0.489954 4.65 1.66995L8.99 6.00995L5.99 9.00995L1.65 4.66995C0.480003 7.09995 0.890003 10.09 2.9 12.1C4.76 13.96 7.48 14.45 9.79 13.58L19.61 23.4L23.32 19.69L13.54 9.89995C14.46 7.55995 13.98 4.79995 12.09 2.90995V2.90995Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-18 pounce-1"
    >
      <div
        class="pounce-16 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
        <svg
          class="pounce-14 pounce-1"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,4 L18,4 L18,2 L16,2 L16,4 L8,4 L8,2 L6,2 L6,4 L5,4 C3.89,4 3.01,4.9 3.01,6 L3,20 C3,21.1 3.89,22 5,22 L19,22 C20.1,22 21,21.1 21,20 L21,6 C21,4.9 20.1,4 19,4 Z M19,20 L5,20 L5,9 L19,9 L19,20 Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-18 pounce-1"
    >
      <div
        class="pounce-16 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
        <svg
          class="pounce-14 pounce-1"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10L12 15L17 10H7Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with invalid option 1`] = `
.pounce-4 {
  position: relative;
}

.pounce-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.pounce-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  color: #f6f6f6;
  background-color: transparent;
  vertical-align: top;
  width: 100%;
  height: 100%;
  border: 0;
  position: relative;
  font-size: 0.875rem;
  font-weight: 500;
}

.pounce-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.pounce-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::placeholder {
  opacity: 0.4;
}

.pounce-0:disabled,
.pounce-0:disabled:focus,
.pounce-0:disabled:hover,
.pounce-0[aria-disabled=true],
.pounce-0[aria-disabled=true]:focus,
.pounce-0[aria-disabled=true]:hover {
  opacity: 1;
  pointer-events: none;
  cursor: default;
}

.pounce-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.pounce-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-16 {
  background-color: transparent;
  min-height: 47px;
  border: 1px solid;
  border-radius: 4px;
  border-color: #d64242;
  position: relative;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-16:disabled,
.pounce-16:disabled:focus,
.pounce-16:disabled:hover,
.pounce-16[aria-disabled=true],
.pounce-16[aria-disabled=true]:focus,
.pounce-16[aria-disabled=true]:hover {
  opacity: 0.3;
  pointer-events: none;
  cursor: default;
}

.pounce-16:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.pounce-2 {
  padding-left: 16px;
  padding-right: 16px;
  color: #d64242;
  top: 0;
  left: 0;
  position: absolute;
  font-size: 0.875rem;
  font-weight: 400;
  pointer-events: none;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-16 pounce-1"
    >
      <div
        class="pounce-14 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="true"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="true"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with required option 1`] = `
.pounce-4 {
  position: relative;
}

.pounce-16 {
  background-color: transparent;
  min-height: 47px;
  border: 1px solid;
  border-radius: 4px;
  border-color: #3e516b;
  position: relative;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-16:hover {
  border-color: #0081c5;
}

.pounce-16:disabled,
.pounce-16:disabled:focus,
.pounce-16:disabled:hover,
.pounce-16[aria-disabled=true],
.pounce-16[aria-disabled=true]:focus,
.pounce-16[aria-disabled=true]:hover {
  opacity: 0.3;
  pointer-events: none;
  cursor: default;
}

.pounce-16:focus-within {
  border-color: #0081c5;
}

.pounce-16:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.pounce-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.pounce-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  color: #f6f6f6;
  background-color: transparent;
  vertical-align: top;
  width: 100%;
  height: 100%;
  border: 0;
  position: relative;
  font-size: 0.875rem;
  font-weight: 500;
}

.pounce-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.pounce-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.pounce-0:focus::placeholder {
  opacity: 0.4;
}

.pounce-0:disabled,
.pounce-0:disabled:focus,
.pounce-0:disabled:hover,
.pounce-0[aria-disabled=true],
.pounce-0[aria-disabled=true]:focus,
.pounce-0[aria-disabled=true]:hover {
  opacity: 1;
  pointer-events: none;
  cursor: default;
}

.pounce-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.pounce-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.pounce-2 {
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  left: 0;
  position: absolute;
  font-size: 0.875rem;
  font-weight: 400;
  pointer-events: none;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="pounce-4 pounce-1"
  >
    <div
      class="pounce-16 pounce-1"
    >
      <div
        class="pounce-14 pounce-1"
      >
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            aria-required="true"
            class="pounce-0 pounce-1"
            id="name-from"
            name="name-from"
            required=""
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="name-from"
          >
            Name
          </label>
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          -
        </div>
        <div
          class="pounce-4 pounce-1"
        >
          <input
            aria-invalid="false"
            aria-required="true"
            class="pounce-0 pounce-1"
            id="surname-to"
            name="surname-to"
            required=""
            type="text"
            value=""
          />
          <label
            class="pounce-2 pounce-1"
            for="surname-to"
          >
            Surname
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
